File: mahaam-api-py/feat/plan/plan_model.py
from uuid import UUID
from datetime import datetime
from feat.user.user_model import User
from pydantic import BaseModel, Field

class Plan(BaseModel):
    id: UUID
    title: str | None = None
    type: str | None = None
    sort_order: int = Field(default=0, serialization_alias="sortOrder")
    starts: datetime | None = None
    ends: datetime | None = None
    done_percent: str | None = Field(default=None, serialization_alias="donePercent")
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")
    updated_at: datetime | None = Field(default=None, serialization_alias="updatedAt")
    members: list[User] | None = Field(default=None, serialization_alias="members")
    is_shared: bool = Field(default=False, serialization_alias="isShared")
    user: User | None = None


class PlanIn(BaseModel):
    id: UUID | None = None
    title: str | None = None
    starts: datetime | None = None
    ends: datetime | None = None


class PlanType:
    MAIN = "Main"
    ARCHIVED = "Archived"

    ALL = [MAIN, ARCHIVED]

File: mahaam-api-java/src/main/java/mahaam/feat/plan/PlanModel.java
package mahaam.feat.plan;

import java.time.Instant;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

import mahaam.feat.user.UserModel.User;

public class PlanModel {

	public static class Plan {
		public UUID id;
		public String title;
		public String type;
		public int sortOrder;
		public Instant starts;
		public Instant ends;
		public String donePercent;
		public Instant createdAt;
		public Instant updatedAt;
		public List<User> members;
		public boolean isShared;
		public User user;
	}

	public static class PlanIn {
		public UUID id;
		public String title;
		public LocalDate starts;
		public LocalDate ends;
	}

	public static class PlanType {
		public static final String MAIN = "Main";
		public static final String ARCHIVED = "Archived";
		public static final List<String> ALL = List.of(MAIN, ARCHIVED);
	}
}
